### PR TITLE
High level API for EdDSA keys

### PR DESCRIPTION
**Pull request by tomato42**
Add EdDSA support to SigningKey, VerifyingKey
Serialisation of them to PEM/PKCS#8 files

### CODE REVIEW

**Review by t184256 — COMMENTED** [review]
Overarching question: why not make new curves (and possibly keys using them) and previously existing ones different subclasses of a common extracted superclass? Seems like their interfaces are pretty different, and the current amount of `isinstance` checks doesn't look great.

File: src/ecdsa/keys.py [hunk 1]
```diff
@@ -306,6 +320,16 @@ def from_string(
         :return: Initialised VerifyingKey object
         :rtype: VerifyingKey
         """
+        if isinstance(curve.curve, CurveEdTw):
+            self = cls(_error__please_use_generate=True)
```
[the remark this answers is not in the document]
Comment (tomato42) [reply src/ecdsa/keys.py:324]: Yes, I'm not a fan of existing API design either, just keeping it for compatibility.

**Comment by tomato42** [comment]
> Overarching question: why not make new curves (and possibly keys using them) and previously existing ones different subclasses of a common extracted superclass? Seems like their interfaces are pretty different, and the current amount of `isinstance` checks doesn't look great.

Because they are stored in the same PKCS#8 files and support the same `hashAndSign()` or `generate()` interface.
While many methods are gated behind `isinstance` checks, not all of them are actually impossible to implement for EdDSA:
 - if support for Ed25519ph is added, then sign() method will be completely valid way to use it
 - `from_secret_exponent()` technically also could be made to work (it would be unexportable as a PKCS#8 file, but you could make signatures with it), etc.

File: src/ecdsa/keys.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (t184256) [reply src/ecdsa/keys.py:324]: My suggestion shouldn't break compatibility. Assuming `_error__please_use_generate` is private, `__init__` could be made into a unconditionally failing thing, while the rest of the constructors could switch over to some form of `__new__` instead.

**Review by t184256 — APPROVED** [review]
Approving, though this is pretty shallow because I lack the insight into the subject matter; it's abstract code to me.

File: src/ecdsa/keys.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (tomato42) [reply src/ecdsa/keys.py:324]: on one hand that assumes that nobody subclassed VerifyingKey or SigningKey
on the other hand, this kind of `__init__` doesn't exactly support subclassing...

in the end I don't see any benefit from changing it

**Pull request merged**